diff -p -up bacula-5.0.3/scripts/Makefile.in.some_scripts_should_be_configuration_files bacula-5.0.3/scripts/Makefile.in
--- bacula-5.0.3/scripts/Makefile.in.some_scripts_should_be_configuration_files	2010-08-05 14:29:51.000000000 +0000
+++ bacula-5.0.3/scripts/Makefile.in	2011-01-08 13:28:35.148591019 +0000
@@ -28,6 +28,7 @@ depend:
 
 installdirs:
 	$(MKDIR) $(DESTDIR)$(sysconfdir)
+	$(MKDIR) $(DESTDIR)$(sysconfdir)/scripts
 	$(MKDIR) $(DESTDIR)$(sbindir)
 	$(MKDIR) $(DESTDIR)$(scriptdir)
 	$(MKDIR) $(DESTDIR)$(mandir)
@@ -42,27 +43,27 @@ install: installdirs
 	$(INSTALL_SCRIPT) bacula-ctl-dir $(DESTDIR)$(scriptdir)/bacula-ctl-dir
 	$(INSTALL_SCRIPT) bacula-ctl-fd $(DESTDIR)$(scriptdir)/bacula-ctl-fd
 	$(INSTALL_SCRIPT) bacula-ctl-sd $(DESTDIR)$(scriptdir)/bacula-ctl-sd
-	@if  test -f ${DESTDIR}${scriptdir}/mtx-changer; then \
+	@if  test -f ${DESTDIR}${sysconfdir}/scripts/mtx-changer; then \
 	   echo "  ==> Saving existing mtx-changer to mtx-changer.old"; \
-	   $(MV) -f ${DESTDIR}${scriptdir}/mtx-changer ${DESTDIR}${scriptdir}/mtx-changer.old; \
+	   $(MV) -f ${DESTDIR}${sysconfdir}/scripts/mtx-changer ${DESTDIR}${sysconfdir}/scripts/mtx-changer.old; \
 	fi
-	$(INSTALL_SCRIPT) mtx-changer $(DESTDIR)$(scriptdir)/mtx-changer
-	@if  test -f ${DESTDIR}${scriptdir}/mtx-changer.conf; then \
+	$(INSTALL_SCRIPT) mtx-changer $(DESTDIR)${sysconfdir}/scripts/mtx-changer
+	@if  test -f ${DESTDIR}${sysconfdir}/scripts/mtx-changer.conf; then \
 	   echo "  ==> Installing mtx-changer.conf to mtx-changer.conf.new"; \
-	   $(INSTALL_SCRIPT) mtx-changer.conf $(DESTDIR)$(scriptdir)/mtx-changer.conf.new; \
+	   $(INSTALL_SCRIPT) mtx-changer.conf $(DESTDIR)${sysconfdir}/scripts/mtx-changer.conf.new; \
 	else \
-	   $(INSTALL_SCRIPT) mtx-changer.conf $(DESTDIR)$(scriptdir)/mtx-changer.conf; \
+	   $(INSTALL_SCRIPT) mtx-changer.conf $(DESTDIR)${sysconfdir}/scripts/mtx-changer.conf; \
 	fi
-	@if  test -f ${DESTDIR}${scriptdir}/disk-changer; then \
+	@if  test -f ${DESTDIR}${sysconfdir}/scripts/disk-changer; then \
 	   echo "  ==> Saving existing disk-changer to disk-changer.old"; \
-	   $(MV) -f ${DESTDIR}${scriptdir}/disk-changer ${DESTDIR}${scriptdir}/disk-changer.old; \
+	   $(MV) -f ${DESTDIR}${sysconfdir}/scripts/disk-changer ${DESTDIR}${sysconfdir}/scripts/disk-changer.old; \
 	fi
-	$(INSTALL_SCRIPT) disk-changer $(DESTDIR)$(scriptdir)/disk-changer
-	@if  test -f ${DESTDIR}${scriptdir}/dvd-handler; then \
+	$(INSTALL_SCRIPT) disk-changer $(DESTDIR)$(sysconfdir)/scripts/disk-changer
+	@if  test -f ${DESTDIR}${sysconfdir}/scripts/dvd-handler; then \
 	   echo "  ==> Saving existing dvd-handler to dvd-handler.old"; \
-	   $(MV) -f ${DESTDIR}${scriptdir}/dvd-handler ${DESTDIR}${scriptdir}/dvd-handler.old; \
+	   $(MV) -f ${DESTDIR}${sysconfdir}/scripts/dvd-handler ${DESTDIR}${sysconfdir}/scripts/dvd-handler.old; \
 	fi
-	$(INSTALL_SCRIPT) dvd-handler $(DESTDIR)$(scriptdir)/dvd-handler
+	$(INSTALL_SCRIPT) dvd-handler $(DESTDIR)${sysconfdir}/scripts/dvd-handler
 	$(INSTALL_DATA)   btraceback.gdb $(DESTDIR)$(scriptdir)/btraceback.gdb
 	$(INSTALL_DATA)   btraceback.dbx $(DESTDIR)$(scriptdir)/btraceback.dbx
 	$(INSTALL_DATA)   btraceback.mdb $(DESTDIR)$(scriptdir)/btraceback.mdb
diff -p -up bacula-5.0.3/scripts/mtx-changer.in.some_scripts_should_be_configuration_files bacula-5.0.3/scripts/mtx-changer.in
--- bacula-5.0.3/scripts/mtx-changer.in.some_scripts_should_be_configuration_files	2010-08-05 14:29:51.000000000 +0000
+++ bacula-5.0.3/scripts/mtx-changer.in	2011-01-08 15:15:11.776587018 +0000
@@ -47,13 +47,13 @@
 #
 
 # source our conf file
-if test ! -f @scriptdir@/mtx-changer.conf ; then
+if test ! -f @sysconfdir@/scripts/mtx-changer.conf ; then
   echo "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
   echo "ERROR: @scriptdir@/mtx-changer.conf file not found!!!!"
   echo "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
   exit 1
 fi
-. @scriptdir@/mtx-changer.conf
+. @sysconfdir@/scripts/mtx-changer.conf
 
 MTX=@MTX@
 
diff -p -up bacula-5.0.3/src/cats/Makefile.in.some_scripts_should_be_configuration_files bacula-5.0.3/src/cats/Makefile.in
--- bacula-5.0.3/src/cats/Makefile.in.some_scripts_should_be_configuration_files	2010-08-05 14:29:51.000000000 +0000
+++ bacula-5.0.3/src/cats/Makefile.in	2011-01-08 13:28:35.148591019 +0000
@@ -156,23 +156,23 @@ install: @LIBTOOL_INSTALL_TARGET@ @INCLU
 	   echo "$(INSTALL_SCRIPT) $$filename $(DESTDIR)$(scriptdir)/$$destname"; \
 	   $(INSTALL_SCRIPT) $$filename $(DESTDIR)$(scriptdir)/$$destname
 	@filename=make_catalog_backup; \
-	   if  test -f $(DESTDIR)$(scriptdir)/$$filename; then \
+	   if  test -f $(DESTDIR)$(sysconfdir)/scripts/$$filename; then \
 	      destname=$$filename.new; \
 	      echo "  ==> Found existing $$filename, installing new file as $$destname"; \
 	   else \
 	      destname=$$filename; \
 	   fi; \
-	   echo "$(INSTALL_SCRIPT) $$filename $(DESTDIR)$(scriptdir)/$$destname"; \
-	   $(INSTALL_SCRIPT) $$filename $(DESTDIR)$(scriptdir)/$$destname
+	   echo "$(INSTALL_SCRIPT) $$filename $(DESTDIR)$(sysconfdir)/scripts/$$destname"; \
+	   $(INSTALL_SCRIPT) $$filename $(DESTDIR)$(sysconfdir)/scripts/$$destname
 	@filename=delete_catalog_backup; \
-	   if  test -f $(DESTDIR)$(scriptdir)/$$filename; then \
+	   if  test -f $(DESTDIR)$(sysconfdir)/scripts/$$filename; then \
 	      destname=$$filename.new; \
 	      echo "  ==> Found existing $$filename, installing new file as $$destname"; \
 	   else \
 	      destname=$$filename; \
 	   fi; \
-	   echo "$(INSTALL_SCRIPT) $$filename $(DESTDIR)$(scriptdir)/$$destname"; \
-	   $(INSTALL_SCRIPT) $$filename $(DESTDIR)$(scriptdir)/$$destname
+	   echo "$(INSTALL_SCRIPT) $$filename $(DESTDIR)$(sysconfdir)/scripts/$$destname"; \
+	   $(INSTALL_SCRIPT) $$filename $(DESTDIR)$(sysconfdir)/scripts/$$destname
 
 uninstall: @LIBTOOL_UNINSTALL_TARGET@ @INCLUDE_UNINSTALL_TARGET@
 	(cd $(DESTDIR)$(scriptdir); $(RMF) create_@DB_TYPE@_database)
diff -p -up bacula-5.0.3/src/dird/Makefile.in.some_scripts_should_be_configuration_files bacula-5.0.3/src/dird/Makefile.in
--- bacula-5.0.3/src/dird/Makefile.in.some_scripts_should_be_configuration_files	2010-08-05 14:29:51.000000000 +0000
+++ bacula-5.0.3/src/dird/Makefile.in	2011-01-08 13:28:35.148591019 +0000
@@ -106,11 +106,11 @@ install: all
 	   fi; \
 	   echo "${INSTALL_CONFIG} $$srcconf ${DESTDIR}${sysconfdir}/$$destconf"; \
 	   ${INSTALL_CONFIG} $$srcconf ${DESTDIR}${sysconfdir}/$$destconf
-	@if  test -f ${DESTDIR}${scriptdir}/query.sql; then \
+	@if  test -f ${DESTDIR}${sysconfdir}/scripts/query.sql; then \
 	   echo "  ==> Saving existing query.sql to query.sql.old"; \
-	   $(MV) -f ${DESTDIR}${scriptdir}/query.sql ${DESTDIR}${scriptdir}/query.sql.old; \
+	   $(MV) -f ${DESTDIR}${sysconfdir}/scripts/query.sql ${DESTDIR}${sysconfdir}/scripts/query.sql.old; \
 	fi 
-	${INSTALL_DATA} query.sql ${DESTDIR}${scriptdir}/query.sql
+	${INSTALL_DATA} query.sql ${DESTDIR}${sysconfdir}/scripts/query.sql
 	@if test -f static-bacula-dir; then \
 	   $(LIBTOOL_INSTALL) $(INSTALL_PROGRAM) static-bacula-dir $(DESTDIR)$(sbindir)/static-bacula-dir; \
 	fi
diff -p -up bacula-5.0.3/src/dird/bacula-dir.conf.in.some_scripts_should_be_configuration_files bacula-5.0.3/src/dird/bacula-dir.conf.in
--- bacula-5.0.3/src/dird/bacula-dir.conf.in.some_scripts_should_be_configuration_files	2011-01-08 13:28:35.128591019 +0000
+++ bacula-5.0.3/src/dird/bacula-dir.conf.in	2011-01-08 13:28:35.148591019 +0000
@@ -15,7 +15,7 @@
 Director {                            # define myself
   Name = @basename@-dir
   DIRport = @dir_port@                # where we listen for UA connections
-  QueryFile = "@scriptdir@/query.sql"
+  QueryFile = "@sysconfdir@/scripts/query.sql"
   WorkingDirectory = "@working_dir@"
   PidDirectory = "@piddir@"
   Maximum Concurrent Jobs = 1
@@ -63,9 +63,9 @@ Job {
   # This creates an ASCII copy of the catalog
   # Arguments to make_catalog_backup.pl are:
   #  make_catalog_backup.pl <catalog-name>
-  RunBeforeJob = "@scriptdir@/make_catalog_backup.pl MyCatalog"
+  RunBeforeJob = "@sysconfdir@/scripts/make_catalog_backup.pl MyCatalog"
   # This deletes the copy of the catalog
-  RunAfterJob  = "@scriptdir@/delete_catalog_backup"
+  RunAfterJob  = "@sysconfdir@/scripts/delete_catalog_backup"
   Write Bootstrap = "@working_dir@/%n.bsr"
   Priority = 11                   # run after main backup
 }
diff -p -up bacula-5.0.3/src/stored/bacula-sd.conf.in.some_scripts_should_be_configuration_files bacula-5.0.3/src/stored/bacula-sd.conf.in
--- bacula-5.0.3/src/stored/bacula-sd.conf.in.some_scripts_should_be_configuration_files	2011-01-08 13:28:35.128591019 +0000
+++ bacula-5.0.3/src/stored/bacula-sd.conf.in	2011-01-08 13:28:35.148591019 +0000
@@ -68,7 +68,7 @@ Device {
 #  Name = Autochanger
 #  Device = Drive-1
 #  Device = Drive-2
-#  Changer Command = "@scriptdir@/mtx-changer %c %o %S %a %d"
+#  Changer Command = "@sysconfdir@/scripts/mtx-changer %c %o %S %a %d"
 #  Changer Device = /dev/sg0
 #}
 
@@ -122,7 +122,7 @@ Device {
 #  RemovableMedia = yes;
 #  RandomAccess = no;
 #  Maximum File Size = 3GB
-## Changer Command = "@scriptdir@/mtx-changer %c %o %S %a %d"
+## Changer Command = "@sysconfdir@/scripts/mtx-changer %c %o %S %a %d"
 ## Changer Device = /dev/sg0
 ## AutoChanger = yes
 #  # Enable the Alert command only if you have the mtx package loaded
@@ -143,7 +143,7 @@ Device {
 #  RemovableMedia = yes;
 #  RandomAccess = no;
 #  Maximum File Size = 4GB
-## Changer Command = "@scriptdir@/mtx-changer %c %o %S %a %d"
+## Changer Command = "@sysconfdir@/scripts/mtx-changer %c %o %S %a %d"
 ## Changer Device = /dev/sg0
 ## AutoChanger = yes
 #  # Enable the Alert command only if you have the mtx package loaded
@@ -164,7 +164,7 @@ Device {
 #  RemovableMedia = yes;
 #  RandomAccess = no;
 #  Maximum File Size = 5GB
-## Changer Command = "@scriptdir@/mtx-changer %c %o %S %a %d"
+## Changer Command = "@sysconfdir@/scripts/mtx-changer %c %o %S %a %d"
 ## Changer Device = /dev/sg0
 ## AutoChanger = yes
 #  # Enable the Alert command only if you have the mtx package loaded
