--- src/wx-console/Makefile.in.link	2011-01-04 04:39:57.000000000 +0000
+++ src/wx-console/Makefile.in	2011-01-04 04:40:53.000000000 +0000
@@ -82,7 +82,7 @@
 	cp bwx-console.conf /Library/Preferences/org.bacula.wxconsole.conf
 
 bwx-console: Makefile $(CONSOBJS) @WIN32@ ../lib/libbaccfg$(DEFAULT_ARCHIVE_TYPE) ../lib/libbac$(DEFAULT_ARCHIVE_TYPE)
-	$(LIBTOOL_LINK) $(CXX) $(CONSOBJS) $(WIN32RES) -o $@ -L../lib $(CONS_LDFLAGS) \
+	$(LIBTOOL_LINK) $(CXX) $(LDFLAGS) $(CONSOBJS) $(WIN32RES) -o $@ -L../lib $(CONS_LDFLAGS) \
 	  -lbaccfg -lbac $(LIBS) $(OPENSSL_LIBS)
 
 win32: bwx-console_private.res
