diff -Naur bacula-2.2.1/src/dird/bacula-dir.conf.in bacula-2.2.1.oden/src/dird/bacula-dir.conf.in
--- bacula-2.2.1/src/dird/bacula-dir.conf.in	2007-05-27 21:30:39.000000000 +0200
+++ bacula-2.2.1.oden/src/dird/bacula-dir.conf.in	2007-09-03 15:40:37.000000000 +0200
@@ -21,6 +21,7 @@
   Maximum Concurrent Jobs = 1
   Password = "@dir_password@"         # Console password
   Messages = Daemon
+  DirAddress = 127.0.0.1
 }
 
 JobDefs {
diff -Naur bacula-2.2.1/src/filed/bacula-fd.conf.in bacula-2.2.1.oden/src/filed/bacula-fd.conf.in
--- bacula-2.2.1/src/filed/bacula-fd.conf.in	2006-03-24 17:37:56.000000000 +0100
+++ bacula-2.2.1.oden/src/filed/bacula-fd.conf.in	2007-09-03 15:40:37.000000000 +0200
@@ -34,6 +34,7 @@
   WorkingDirectory = @working_dir@
   Pid Directory = @piddir@
   Maximum Concurrent Jobs = 20
+  FDAddress = 127.0.0.1
 }
 
 # Send all messages except skipped files back to Director
diff -Naur bacula-2.2.1/src/stored/bacula-sd.conf.in bacula-2.2.1.oden/src/stored/bacula-sd.conf.in
--- bacula-2.2.1/src/stored/bacula-sd.conf.in	2007-04-23 18:31:14.000000000 +0200
+++ bacula-2.2.1.oden/src/stored/bacula-sd.conf.in	2007-09-03 15:40:37.000000000 +0200
@@ -16,6 +16,7 @@
   WorkingDirectory = "@working_dir@"
   Pid Directory = "@piddir@"
   Maximum Concurrent Jobs = 20
+  SDAddress = 127.0.0.1
 }
 
 #
diff -Naur bacula-2.2.1/src/stored/stored.conf.in bacula-2.2.1.oden/src/stored/stored.conf.in
--- bacula-2.2.1/src/stored/stored.conf.in	2002-04-23 21:48:51.000000000 +0200
+++ bacula-2.2.1.oden/src/stored/stored.conf.in	2007-09-03 15:40:37.000000000 +0200
@@ -13,6 +13,7 @@
   WorkingDirectory = "@working_dir@"
   Pid Directory = "@piddir@"
   Subsys Directory = "@subsysdir@"
+  SDAddress = 127.0.0.1
 }
 
 #
