diff -p -up bacula-5.0.3/autoconf/bacula-macros/db.m4.mysql bacula-5.0.3/autoconf/bacula-macros/db.m4
--- bacula-5.0.3/autoconf/bacula-macros/db.m4.mysql	2010-08-05 14:29:51.000000000 +0000
+++ bacula-5.0.3/autoconf/bacula-macros/db.m4	2011-01-06 20:02:46.699341001 +0000
@@ -139,44 +139,44 @@ AC_HELP_STRING([--with-dbi-driver@<:@=DR
            db_prog="mysql"
            if test -f /usr/local/mysql/bin/mysql; then
               SQL_BINDIR=/usr/local/mysql/bin
-              if test -f /usr/local/mysql/lib64/mysql/libmysqlclient_r.a \
-                 -o -f /usr/local/mysql/lib64/mysql/libmysqlclient_r.so; then
+              if test -f /usr/local/mysql/lib64/mysql/libmysqlclient.a \
+                 -o -f /usr/local/mysql/lib64/mysql/libmysqlclient.so; then
                  SQL_LIBDIR=/usr/local/mysql/lib64/mysql
               else
                  SQL_LIBDIR=/usr/local/mysql/lib/mysql
               fi
            elif test -f /usr/bin/mysql; then
               SQL_BINDIR=/usr/bin
-              if test -f /usr/lib64/mysql/libmysqlclient_r.a \
-                  -o -f /usr/lib64/mysql/libmysqlclient_r.so; then  
+              if test -f /usr/lib64/mysql/libmysqlclient.a \
+                  -o -f /usr/lib64/mysql/libmysqlclient.so; then  
                   SQL_LIBDIR=/usr/lib64/mysql
-              elif test -f /usr/lib/mysql/libmysqlclient_r.a \
-                  -o -f /usr/lib/mysql/libmysqlclient_r.so; then
+              elif test -f /usr/lib/mysql/libmysqlclient.a \
+                  -o -f /usr/lib/mysql/libmysqlclient.so; then
                   SQL_LIBDIR=/usr/lib/mysql
               else
                   SQL_LIBDIR=/usr/lib
               fi
            elif test -f /usr/local/bin/mysql; then
               SQL_BINDIR=/usr/local/bin
-              if test -f /usr/local/lib64/mysql/libmysqlclient_r.a \
-                  -o -f /usr/local/lib64/mysql/libmysqlclient_r.so; then  
+              if test -f /usr/local/lib64/mysql/libmysqlclient.a \
+                  -o -f /usr/local/lib64/mysql/libmysqlclient.so; then  
                   SQL_LIBDIR=/usr/local/lib64/mysql
-              elif test -f /usr/local/lib/mysql/libmysqlclient_r.a \
-                  -o -f /usr/local/lib/mysql/libmysqlclient_r.so; then
+              elif test -f /usr/local/lib/mysql/libmysqlclient.a \
+                  -o -f /usr/local/lib/mysql/libmysqlclient.so; then
                   SQL_LIBDIR=/usr/local/lib/mysql
               else
                   SQL_LIBDIR=/usr/local/lib
               fi
            elif test -f $withval/bin/mysql; then
               SQL_BINDIR=$withval/bin
-              if test -f $withval/lib64/mysql/libmysqlclient_r.a \
-                  -o -f $withval/lib64/mysql/libmysqlclient_r.so; then
+              if test -f $withval/lib64/mysql/libmysqlclient.a \
+                  -o -f $withval/lib64/mysql/libmysqlclient.so; then
                   SQL_LIBDIR=$withval/lib64/mysql
-              elif test -f $withval/lib64/libmysqlclient_r.a \
-                  -o -f $withval/lib64/libmysqlclient_r.so; then
+              elif test -f $withval/lib64/libmysqlclient.a \
+                  -o -f $withval/lib64/libmysqlclient.so; then
                   SQL_LIBDIR=$withval/lib64
-              elif test -f $withval/lib/libmysqlclient_r.a \
-                  -o -f $withval/lib/libmysqlclient_r.so; then
+              elif test -f $withval/lib/libmysqlclient.a \
+                  -o -f $withval/lib/libmysqlclient.so; then
                   SQL_LIBDIR=$withval/lib/
               else
                   SQL_LIBDIR=$withval/lib/mysql
@@ -185,10 +185,10 @@ AC_HELP_STRING([--with-dbi-driver@<:@=DR
               AC_MSG_RESULT(no)
               AC_MSG_ERROR(Unable to find mysql in standard locations)
            fi
-           if test -f $SQL_LIBDIR/libmysqlclient_r.so; then
-              DB_PROG_LIB=$SQL_LIBDIR/libmysqlclient_r.so
+           if test -f $SQL_LIBDIR/libmysqlclient.so; then
+              DB_PROG_LIB=$SQL_LIBDIR/libmysqlclient.so
            else
-              DB_PROG_LIB=$SQL_LIBDIR/libmysqlclient_r.a
+              DB_PROG_LIB=$SQL_LIBDIR/libmysqlclient.a
            fi
         ;;
         "postgresql")
@@ -330,8 +330,8 @@ AC_HELP_STRING([--with-mysql@<:@=DIR@:>@
         if test "$withval" = "yes"; then
            if test -f /usr/local/mysql/include/mysql/mysql.h; then
                    MYSQL_INCDIR=/usr/local/mysql/include/mysql
-                   if test -f /usr/local/mysql/lib64/mysql/libmysqlclient_r.a \
-                        -o -f /usr/local/mysql/lib64/mysql/libmysqlclient_r.so; then
+                   if test -f /usr/local/mysql/lib64/mysql/libmysqlclient.a \
+                        -o -f /usr/local/mysql/lib64/mysql/libmysqlclient.so; then
                            MYSQL_LIBDIR=/usr/local/mysql/lib64/mysql
                    else
                            MYSQL_LIBDIR=/usr/local/mysql/lib/mysql
@@ -339,14 +339,14 @@ AC_HELP_STRING([--with-mysql@<:@=DIR@:>@
                    MYSQL_BINDIR=/usr/local/mysql/bin
            elif test -f /usr/include/mysql/mysql.h; then
                    MYSQL_INCDIR=/usr/include/mysql
-                   if test -f /usr/lib64/mysql/libmysqlclient_r.a \
-                        -o -f /usr/lib64/mysql/libmysqlclient_r.so; then  
+                   if test -f /usr/lib64/mysql/libmysqlclient.a \
+                        -o -f /usr/lib64/mysql/libmysqlclient.so; then  
                            MYSQL_LIBDIR=/usr/lib64/mysql
-                   elif test -f /usr/lib64/libmysqlclient_r.a \
-                        -o -f /usr/lib64/libmysqlclient_r.so; then
+                   elif test -f /usr/lib64/libmysqlclient.a \
+                        -o -f /usr/lib64/libmysqlclient.so; then
                            MYSQL_LIBDIR=/usr/lib64
-                   elif test -f /usr/lib/mysql/libmysqlclient_r.a \
-                          -o -f /usr/lib/mysql/libmysqlclient_r.so; then
+                   elif test -f /usr/lib/mysql/libmysqlclient.a \
+                          -o -f /usr/lib/mysql/libmysqlclient.so; then
                            MYSQL_LIBDIR=/usr/lib/mysql
                    else
                            MYSQL_LIBDIR=/usr/lib
@@ -354,8 +354,8 @@ AC_HELP_STRING([--with-mysql@<:@=DIR@:>@
                    MYSQL_BINDIR=/usr/bin
            elif test -f /usr/include/mysql.h; then
                    MYSQL_INCDIR=/usr/include
-                   if test -f /usr/lib64/libmysqlclient_r.a \
-                        -o -f /usr/lib64/libmysqlclient_r.so; then
+                   if test -f /usr/lib64/libmysqlclient.a \
+                        -o -f /usr/lib64/libmysqlclient.so; then
                            MYSQL_LIBDIR=/usr/lib64
                    else
                            MYSQL_LIBDIR=/usr/lib
@@ -363,8 +363,8 @@ AC_HELP_STRING([--with-mysql@<:@=DIR@:>@
                    MYSQL_BINDIR=/usr/bin
            elif test -f /usr/local/include/mysql/mysql.h; then
                    MYSQL_INCDIR=/usr/local/include/mysql
-                   if test -f /usr/local/lib64/mysql/libmysqlclient_r.a \
-                        -o -f /usr/local/lib64/mysql/libmysqlclient_r.so; then
+                   if test -f /usr/local/lib64/mysql/libmysqlclient.a \
+                        -o -f /usr/local/lib64/mysql/libmysqlclient.so; then
                            MYSQL_LIBDIR=/usr/local/lib64/mysql
                    else
                            MYSQL_LIBDIR=/usr/local/lib/mysql
@@ -372,8 +372,8 @@ AC_HELP_STRING([--with-mysql@<:@=DIR@:>@
                    MYSQL_BINDIR=/usr/local/bin
            elif test -f /usr/local/include/mysql.h; then
                    MYSQL_INCDIR=/usr/local/include
-                   if test -f /usr/local/lib64/libmysqlclient_r.a \
-                        -o -f /usr/local/lib64/libmysqlclient_r.so; then
+                   if test -f /usr/local/lib64/libmysqlclient.a \
+                        -o -f /usr/local/lib64/libmysqlclient.so; then
                            MYSQL_LIBDIR=/usr/local/lib64
                    else
                            MYSQL_LIBDIR=/usr/local/lib
@@ -386,14 +386,14 @@ AC_HELP_STRING([--with-mysql@<:@=DIR@:>@
         else
            if test -f $withval/include/mysql/mysql.h; then
               MYSQL_INCDIR=$withval/include/mysql
-              if test -f $withval/lib64/mysql/libmysqlclient_r.a \
-                   -o -f $withval/lib64/mysql/libmysqlclient_r.so; then
+              if test -f $withval/lib64/mysql/libmysqlclient.a \
+                   -o -f $withval/lib64/mysql/libmysqlclient.so; then
                  MYSQL_LIBDIR=$withval/lib64/mysql
-              elif test -f $withval/lib64/libmysqlclient_r.a \
-                   -o -f $withval/lib64/libmysqlclient_r.so; then
+              elif test -f $withval/lib64/libmysqlclient.a \
+                   -o -f $withval/lib64/libmysqlclient.so; then
                  MYSQL_LIBDIR=$withval/lib64
-              elif test -f $withval/lib/libmysqlclient_r.a \
-                   -o -f $withval/lib/libmysqlclient_r.so; then
+              elif test -f $withval/lib/libmysqlclient.a \
+                   -o -f $withval/lib/libmysqlclient.so; then
                  MYSQL_LIBDIR=$withval/lib
               else
                  MYSQL_LIBDIR=$withval/lib/mysql
@@ -401,8 +401,8 @@ AC_HELP_STRING([--with-mysql@<:@=DIR@:>@
               MYSQL_BINDIR=$withval/bin
            elif test -f $withval/include/mysql.h; then
               MYSQL_INCDIR=$withval/include
-              if test -f $withval/lib64/libmysqlclient_r.a \
-                   -o -f $withval/lib64/libmysqlclient_r.so; then
+              if test -f $withval/lib64/libmysqlclient.a \
+                   -o -f $withval/lib64/libmysqlclient.so; then
                  MYSQL_LIBDIR=$withval/lib64
               else
                  MYSQL_LIBDIR=$withval/lib
@@ -414,13 +414,13 @@ AC_HELP_STRING([--with-mysql@<:@=DIR@:>@
            fi
         fi
     SQL_INCLUDE=-I$MYSQL_INCDIR
-    if test -f $MYSQL_LIBDIR/libmysqlclient_r.a \
-         -o -f $MYSQL_LIBDIR/libmysqlclient_r.so; then
-       SQL_LFLAGS="-L$MYSQL_LIBDIR -lmysqlclient_r -lz"
+    if test -f $MYSQL_LIBDIR/libmysqlclient.a \
+         -o -f $MYSQL_LIBDIR/libmysqlclient.so; then
+       SQL_LFLAGS="-L$MYSQL_LIBDIR -lmysqlclient -lz"
        AC_DEFINE(HAVE_THREAD_SAFE_MYSQL)
     fi
     SQL_BINDIR=$MYSQL_BINDIR
-    SQL_LIB=$MYSQL_LIBDIR/libmysqlclient_r.a
+    SQL_LIB=$MYSQL_LIBDIR/libmysqlclient.a
 
     AC_DEFINE(HAVE_MYSQL, 1, [Set if you have an MySQL Database])
     AC_MSG_RESULT(yes)
diff -p -up bacula-5.0.3/configure.mysql bacula-5.0.3/configure
--- bacula-5.0.3/configure.mysql	2011-01-06 20:02:46.679341001 +0000
+++ bacula-5.0.3/configure	2011-01-06 20:08:36.815341002 +0000
@@ -22962,8 +22962,8 @@ if test "${with_mysql+set}" = set; then
         if test "$withval" = "yes"; then
            if test -f /usr/local/mysql/include/mysql/mysql.h; then
                    MYSQL_INCDIR=/usr/local/mysql/include/mysql
-                   if test -f /usr/local/mysql/lib64/mysql/libmysqlclient_r.a \
-                        -o -f /usr/local/mysql/lib64/mysql/libmysqlclient_r.so; then
+                   if test -f /usr/local/mysql/lib64/mysql/libmysqlclient.a \
+                        -o -f /usr/local/mysql/lib64/mysql/libmysqlclient.so; then
                            MYSQL_LIBDIR=/usr/local/mysql/lib64/mysql
                    else
                            MYSQL_LIBDIR=/usr/local/mysql/lib/mysql
@@ -22971,14 +22971,14 @@ if test "${with_mysql+set}" = set; then
                    MYSQL_BINDIR=/usr/local/mysql/bin
            elif test -f /usr/include/mysql/mysql.h; then
                    MYSQL_INCDIR=/usr/include/mysql
-                   if test -f /usr/lib64/mysql/libmysqlclient_r.a \
-                        -o -f /usr/lib64/mysql/libmysqlclient_r.so; then
+                   if test -f /usr/lib64/mysql/libmysqlclient.a \
+                        -o -f /usr/lib64/mysql/libmysqlclient.so; then
                            MYSQL_LIBDIR=/usr/lib64/mysql
-                   elif test -f /usr/lib64/libmysqlclient_r.a \
-                        -o -f /usr/lib64/libmysqlclient_r.so; then
+                   elif test -f /usr/lib64/libmysqlclient.a \
+                        -o -f /usr/lib64/libmysqlclient.so; then
                            MYSQL_LIBDIR=/usr/lib64
-                   elif test -f /usr/lib/mysql/libmysqlclient_r.a \
-                          -o -f /usr/lib/mysql/libmysqlclient_r.so; then
+                   elif test -f /usr/lib/mysql/libmysqlclient.a \
+                          -o -f /usr/lib/mysql/libmysqlclient.so; then
                            MYSQL_LIBDIR=/usr/lib/mysql
                    else
                            MYSQL_LIBDIR=/usr/lib
@@ -22986,8 +22986,8 @@ if test "${with_mysql+set}" = set; then
                    MYSQL_BINDIR=/usr/bin
            elif test -f /usr/include/mysql.h; then
                    MYSQL_INCDIR=/usr/include
-                   if test -f /usr/lib64/libmysqlclient_r.a \
-                        -o -f /usr/lib64/libmysqlclient_r.so; then
+                   if test -f /usr/lib64/libmysqlclient.a \
+                        -o -f /usr/lib64/libmysqlclient.so; then
                            MYSQL_LIBDIR=/usr/lib64
                    else
                            MYSQL_LIBDIR=/usr/lib
@@ -22995,8 +22995,8 @@ if test "${with_mysql+set}" = set; then
                    MYSQL_BINDIR=/usr/bin
            elif test -f /usr/local/include/mysql/mysql.h; then
                    MYSQL_INCDIR=/usr/local/include/mysql
-                   if test -f /usr/local/lib64/mysql/libmysqlclient_r.a \
-                        -o -f /usr/local/lib64/mysql/libmysqlclient_r.so; then
+                   if test -f /usr/local/lib64/mysql/libmysqlclient.a \
+                        -o -f /usr/local/lib64/mysql/libmysqlclient.so; then
                            MYSQL_LIBDIR=/usr/local/lib64/mysql
                    else
                            MYSQL_LIBDIR=/usr/local/lib/mysql
@@ -23004,8 +23004,8 @@ if test "${with_mysql+set}" = set; then
                    MYSQL_BINDIR=/usr/local/bin
            elif test -f /usr/local/include/mysql.h; then
                    MYSQL_INCDIR=/usr/local/include
-                   if test -f /usr/local/lib64/libmysqlclient_r.a \
-                        -o -f /usr/local/lib64/libmysqlclient_r.so; then
+                   if test -f /usr/local/lib64/libmysqlclient.a \
+                        -o -f /usr/local/lib64/libmysqlclient.so; then
                            MYSQL_LIBDIR=/usr/local/lib64
                    else
                            MYSQL_LIBDIR=/usr/local/lib
@@ -23019,14 +23019,14 @@ $as_echo "no" >&6; }
         else
            if test -f $withval/include/mysql/mysql.h; then
               MYSQL_INCDIR=$withval/include/mysql
-              if test -f $withval/lib64/mysql/libmysqlclient_r.a \
-                   -o -f $withval/lib64/mysql/libmysqlclient_r.so; then
+              if test -f $withval/lib64/mysql/libmysqlclient.a \
+                   -o -f $withval/lib64/mysql/libmysqlclient.so; then
                  MYSQL_LIBDIR=$withval/lib64/mysql
-              elif test -f $withval/lib64/libmysqlclient_r.a \
-                   -o -f $withval/lib64/libmysqlclient_r.so; then
+              elif test -f $withval/lib64/libmysqlclient.a \
+                   -o -f $withval/lib64/libmysqlclient.so; then
                  MYSQL_LIBDIR=$withval/lib64
-              elif test -f $withval/lib/libmysqlclient_r.a \
-                   -o -f $withval/lib/libmysqlclient_r.so; then
+              elif test -f $withval/lib/libmysqlclient.a \
+                   -o -f $withval/lib/libmysqlclient.so; then
                  MYSQL_LIBDIR=$withval/lib
               else
                  MYSQL_LIBDIR=$withval/lib/mysql
@@ -23034,8 +23034,8 @@ $as_echo "no" >&6; }
               MYSQL_BINDIR=$withval/bin
            elif test -f $withval/include/mysql.h; then
               MYSQL_INCDIR=$withval/include
-              if test -f $withval/lib64/libmysqlclient_r.a \
-                   -o -f $withval/lib64/libmysqlclient_r.so; then
+              if test -f $withval/lib64/libmysqlclient.a \
+                   -o -f $withval/lib64/libmysqlclient.so; then
                  MYSQL_LIBDIR=$withval/lib64
               else
                  MYSQL_LIBDIR=$withval/lib
@@ -23048,14 +23048,14 @@ $as_echo "no" >&6; }
            fi
         fi
     SQL_INCLUDE=-I$MYSQL_INCDIR
-    if test -f $MYSQL_LIBDIR/libmysqlclient_r.a \
-         -o -f $MYSQL_LIBDIR/libmysqlclient_r.so; then
-       SQL_LFLAGS="-L$MYSQL_LIBDIR -lmysqlclient_r -lz"
+    if test -f $MYSQL_LIBDIR/libmysqlclient.a \
+         -o -f $MYSQL_LIBDIR/libmysqlclient.so; then
+       SQL_LFLAGS="-L$MYSQL_LIBDIR -lmysqlclient -lz"
        $as_echo "#define HAVE_THREAD_SAFE_MYSQL 1" >>confdefs.h
 
     fi
     SQL_BINDIR=$MYSQL_BINDIR
-    SQL_LIB=$MYSQL_LIBDIR/libmysqlclient_r.a
+    SQL_LIB=$MYSQL_LIBDIR/libmysqlclient.a
 
 
 $as_echo "#define HAVE_MYSQL 1" >>confdefs.h
@@ -23496,44 +23496,44 @@ if test "${with_dbi_driver+set}" = set;
            db_prog="mysql"
            if test -f /usr/local/mysql/bin/mysql; then
               SQL_BINDIR=/usr/local/mysql/bin
-              if test -f /usr/local/mysql/lib64/mysql/libmysqlclient_r.a \
-                 -o -f /usr/local/mysql/lib64/mysql/libmysqlclient_r.so; then
+              if test -f /usr/local/mysql/lib64/mysql/libmysqlclient.a \
+                 -o -f /usr/local/mysql/lib64/mysql/libmysqlclient.so; then
                  SQL_LIBDIR=/usr/local/mysql/lib64/mysql
               else
                  SQL_LIBDIR=/usr/local/mysql/lib/mysql
               fi
            elif test -f /usr/bin/mysql; then
               SQL_BINDIR=/usr/bin
-              if test -f /usr/lib64/mysql/libmysqlclient_r.a \
-                  -o -f /usr/lib64/mysql/libmysqlclient_r.so; then
+              if test -f /usr/lib64/mysql/libmysqlclient.a \
+                  -o -f /usr/lib64/mysql/libmysqlclient.so; then
                   SQL_LIBDIR=/usr/lib64/mysql
-              elif test -f /usr/lib/mysql/libmysqlclient_r.a \
-                  -o -f /usr/lib/mysql/libmysqlclient_r.so; then
+              elif test -f /usr/lib/mysql/libmysqlclient.a \
+                  -o -f /usr/lib/mysql/libmysqlclient.so; then
                   SQL_LIBDIR=/usr/lib/mysql
               else
                   SQL_LIBDIR=/usr/lib
               fi
            elif test -f /usr/local/bin/mysql; then
               SQL_BINDIR=/usr/local/bin
-              if test -f /usr/local/lib64/mysql/libmysqlclient_r.a \
-                  -o -f /usr/local/lib64/mysql/libmysqlclient_r.so; then
+              if test -f /usr/local/lib64/mysql/libmysqlclient.a \
+                  -o -f /usr/local/lib64/mysql/libmysqlclient.so; then
                   SQL_LIBDIR=/usr/local/lib64/mysql
-              elif test -f /usr/local/lib/mysql/libmysqlclient_r.a \
-                  -o -f /usr/local/lib/mysql/libmysqlclient_r.so; then
+              elif test -f /usr/local/lib/mysql/libmysqlclient.a \
+                  -o -f /usr/local/lib/mysql/libmysqlclient.so; then
                   SQL_LIBDIR=/usr/local/lib/mysql
               else
                   SQL_LIBDIR=/usr/local/lib
               fi
            elif test -f $withval/bin/mysql; then
               SQL_BINDIR=$withval/bin
-              if test -f $withval/lib64/mysql/libmysqlclient_r.a \
-                  -o -f $withval/lib64/mysql/libmysqlclient_r.so; then
+              if test -f $withval/lib64/mysql/libmysqlclient.a \
+                  -o -f $withval/lib64/mysql/libmysqlclient.so; then
                   SQL_LIBDIR=$withval/lib64/mysql
-              elif test -f $withval/lib64/libmysqlclient_r.a \
-                  -o -f $withval/lib64/libmysqlclient_r.so; then
+              elif test -f $withval/lib64/libmysqlclient.a \
+                  -o -f $withval/lib64/libmysqlclient.so; then
                   SQL_LIBDIR=$withval/lib64
-              elif test -f $withval/lib/libmysqlclient_r.a \
-                  -o -f $withval/lib/libmysqlclient_r.so; then
+              elif test -f $withval/lib/libmysqlclient.a \
+                  -o -f $withval/lib/libmysqlclient.so; then
                   SQL_LIBDIR=$withval/lib/
               else
                   SQL_LIBDIR=$withval/lib/mysql
@@ -23543,10 +23543,10 @@ if test "${with_dbi_driver+set}" = set;
 $as_echo "no" >&6; }
               as_fn_error "Unable to find mysql in standard locations" "$LINENO" 5
            fi
-           if test -f $SQL_LIBDIR/libmysqlclient_r.so; then
-              DB_PROG_LIB=$SQL_LIBDIR/libmysqlclient_r.so
+           if test -f $SQL_LIBDIR/libmysqlclient.so; then
+              DB_PROG_LIB=$SQL_LIBDIR/libmysqlclient.so
            else
-              DB_PROG_LIB=$SQL_LIBDIR/libmysqlclient_r.a
+              DB_PROG_LIB=$SQL_LIBDIR/libmysqlclient.a
            fi
         ;;
         "postgresql")
@@ -23711,7 +23711,7 @@ fi
 
 if test x$support_batch_insert = xno; then
   if test x$DB_TYPE = xmysql; then
-    A=`test -f $MYSQL_LIBDIR/libmysqlclient_r.so && nm -D $MYSQL_LIBDIR/libmysqlclient_r.so | grep pthread_mutex_lock`
+    A=`test -f $MYSQL_LIBDIR/libmysqlclient.so && nm -D $MYSQL_LIBDIR/libmysqlclient.so | grep pthread_mutex_lock`
     pkg=$?
     if test $pkg = 0; then
       support_batch_insert=yes
diff -p -up bacula-5.0.3/src/cats/mysql.c.mysql bacula-5.0.3/src/cats/mysql.c
--- bacula-5.0.3/src/cats/mysql.c.mysql	2010-08-05 14:29:51.000000000 +0000
+++ bacula-5.0.3/src/cats/mysql.c	2011-01-06 20:02:46.699341001 +0000
@@ -292,7 +292,7 @@ void db_check_backend_thread_safe()
 void db_thread_cleanup()
 { 
 #ifndef HAVE_WIN32
-   my_thread_end();
+   mysql_thread_end();
 #endif
 }
 
