diff -p -up bacula-5.0.3/src/stored/bacula-sd.conf.in.sudo_sd bacula-5.0.3/src/stored/bacula-sd.conf.in
--- bacula-5.0.3/src/stored/bacula-sd.conf.in.sudo_sd	2011-01-08 13:28:35.000000000 +0000
+++ bacula-5.0.3/src/stored/bacula-sd.conf.in	2011-01-08 15:22:04.332585883 +0000
@@ -91,7 +91,7 @@ Device {
 #  #
 #  Alert Command = "sh -c 'tapeinfo -f %c |grep TapeAlert|cat'"
 #  If you have smartctl, enable this, it has more info than tapeinfo 
-#  Alert Command = "sh -c 'smartctl -H -l error %c'"  
+#  Alert Command = "sh -c 'sudo smartctl -d scsi -H -l error %a'"
 #}
 
 #Device {
@@ -107,7 +107,7 @@ Device {
 #  # Enable the Alert command only if you have the mtx package loaded
 #  Alert Command = "sh -c 'tapeinfo -f %c |grep TapeAlert|cat'"
 #  If you have smartctl, enable this, it has more info than tapeinfo 
-#  Alert Command = "sh -c 'smartctl -H -l error %c'"  
+#  Alert Command = "sh -c 'sudo smartctl -d scsi -H -l error %a'"
 #}
 
 #
@@ -128,7 +128,7 @@ Device {
 #  # Enable the Alert command only if you have the mtx package loaded
 ## Alert Command = "sh -c 'tapeinfo -f %c |grep TapeAlert|cat'"
 ## If you have smartctl, enable this, it has more info than tapeinfo 
-## Alert Command = "sh -c 'smartctl -H -l error %c'"  
+## Alert Command = "sh -c 'sudo smartctl -d scsi -H -l error %a'"
 #}
 
 #
@@ -149,7 +149,7 @@ Device {
 #  # Enable the Alert command only if you have the mtx package loaded
 ## Alert Command = "sh -c 'tapeinfo -f %c |grep TapeAlert|cat'"
 ## If you have smartctl, enable this, it has more info than tapeinfo 
-## Alert Command = "sh -c 'smartctl -H -l error %c'"  
+## Alert Command = "sh -c 'sudo smartctl -d scsi -H -l error %a'"
 #}
 
 #
@@ -170,7 +170,7 @@ Device {
 #  # Enable the Alert command only if you have the mtx package loaded
 ## Alert Command = "sh -c 'tapeinfo -f %c |grep TapeAlert|cat'"
 ## If you have smartctl, enable this, it has more info than tapeinfo 
-## Alert Command = "sh -c 'smartctl -H -l error %c'"  
+## Alert Command = "sh -c 'sudo smartctl -d scsi -H -l error %a'"
 #}
 
 
@@ -193,7 +193,7 @@ Device {
 #  Fast Forward Space File = no
 #  TWO EOF = yes
 #  If you have smartctl, enable this, it has more info than tapeinfo 
-#  Alert Command = "sh -c 'smartctl -H -l error %c'"  
+#  Alert Command = "sh -c 'sudo smartctl -d scsi -H -l error %a'"
 #}
 
 # 
